### PR TITLE
feat(change-user-source-organization)

### PR DESCRIPTION
**Pull request by nevobonneh**
Add option to change user source organization on users management table
[1216](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1216/)

### CODE REVIEW

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts [hunk 1]
```diff
@@ -215,6 +215,37 @@ const useUsersManagement = ({ page, rowsPerPage, cellNameSort, setPage }: useUse
         });
     }
 
+    const setUserSourceOrganization = (sourceOrganization: string, userId: string) : Promise<any> => {
+        const setUpdateSourcesOrganizationLogger = logger.setup('Updating user source organization');
+        setUpdateSourcesOrganizationLogger.info('send request to server for updating user source organization', Severity.LOW);
+        return axios.post('users/updateSourceOrganizationById', {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/UsersManagement/useUsersManagement.ts:221]: why do you need to return the Promise?

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx [hunk 2]
```diff
@@ -81,6 +82,31 @@ const UsersManagement: React.FC = () => {
             case UsersManagementTableHeadersNames.DESK: {
                 return row[cellName] ? row[cellName] : noDeskAssignment
             }
+            case UsersManagementTableHeadersNames.SOURCE_ORGANIZATION : {
+                return (
+                    <Autocomplete
+                        options={sourcesOrganization}
+                        getOptionLabel={(option) => {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/UsersManagement/UsersManagement.tsx:89]: you can delete the curly braces and make it one line only

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx [hunk 3]
```diff
@@ -81,6 +82,31 @@ const UsersManagement: React.FC = () => {
...
+                            return option.displayName ? option.displayName : option
+                        }}
+                        getOptionSelected={(option, value) => {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/UsersManagement/UsersManagement.tsx:92]: same

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts [hunk 4]
```diff
@@ -215,6 +215,37 @@ const useUsersManagement = ({ page, rowsPerPage, cellNameSort, setPage }: useUse
...
+        }).catch((error) => {
+            alertError('לא הצלחנו לעדכן את מסגרת המשתמש');
+            setUpdateSourcesOrganizationLogger.error(`error in updating user source organization ${error}`, Severity.HIGH);
+        });
+    }
+
+    const deleteUserSourceOrganization = (userId: string) : Promise<any> => {
+        const setDeleteSourcesOrganizationLogger = logger.setup('Deleting user source organization');
+        setDeleteSourcesOrganizationLogger.info('send request to server for deleting user source organization', Severity.LOW);
+        return axios.post('users/deleteSourceOrganizationById', {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/UsersManagement/useUsersManagement.ts:237]: same

File: server/src/ClientToDBAPI/UsersRoute/mainRoute.ts [hunk 5]
```diff
@@ -42,6 +42,57 @@ usersRoute.get('/userActivityStatus', (request: Request, response: Response) =>
...
+        sourceOrganization : request.body.sourceOrganization
+    };
+    const updateSourceOrganizationLogger = logger.setup({
+        workflow: 'Updating user source organization',
+        user: response.locals.user.id,
+    });
+    updateSourceOrganizationLogger.info(`make the graphql API request with parameters ${JSON.stringify(updateSourceOrganizationVariables)}`, Severity.LOW);
+    graphqlRequest(UPDATE_SOURCE_ORGANIZATION, response.locals, updateSourceOrganizationVariables)
+        .then((result: any) => {
+            if (result.data.updateUserById) {
```
Comment (ItayBenMoshe) [inline server/src/ClientToDBAPI/UsersRoute/mainRoute.ts:57]: check for !result.errors also

File: server/src/ClientToDBAPI/UsersRoute/mainRoute.ts [hunk 6]
```diff
@@ -42,6 +42,57 @@ usersRoute.get('/userActivityStatus', (request: Request, response: Response) =>
...
+                updateSourceOrganizationLogger.info('got response from the DB', Severity.LOW);
+                response.send(result.data.updateUserById.user);
+            }
+            else {
+                updateSourceOrganizationLogger.error('didnt get data from the DB', Severity.HIGH);
```
Comment (ItayBenMoshe) [inline server/src/ClientToDBAPI/UsersRoute/mainRoute.ts:62]: log result.errors[0]?.message

File: server/src/ClientToDBAPI/UsersRoute/mainRoute.ts [hunk 7]
```diff
@@ -42,6 +42,57 @@ usersRoute.get('/userActivityStatus', (request: Request, response: Response) =>
...
+usersRoute.post('/deleteSourceOrganizationById', adminMiddleWare, (request: Request, response: Response) => {
+    const id = request.body.userId;
+    const deleteSourceOrganizationLogger = logger.setup({
+        workflow: 'Deleting user source organization',
+        user: response.locals.user.id,
+    });
+    deleteSourceOrganizationLogger.info(`make the graphql API request with parameter ${id}`, Severity.LOW);
+    graphqlRequest(DELETE_SOURCE_ORGANIZATION, response.locals, {id})
+        .then((result: any) => {
+            if (result.data.updateUserById) {
```
Comment (ItayBenMoshe) [inline server/src/ClientToDBAPI/UsersRoute/mainRoute.ts:81]: same

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx [hunk 8]
```diff
@@ -81,6 +82,38 @@ const UsersManagement: React.FC = () => {
...
+                                vertical: "top",
+                                horizontal: "left"
+                            },
+                            getContentAnchorEl: null
+                        }}
+                        variant='outlined'
+                        className={classes.sourceOrganization}
+                        label={sourceOrganizationLabel}
+                        value={row[cellName]}
+                        onChange={(event: any) => setUserSourceOrganization(event.target.value, row[UsersManagementTableHeadersNames.MABAR_USER_NAME])}
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/UsersManagement/UsersManagement.tsx:103]: change any type

**Review by ItayBenMoshe — APPROVED** [review]

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx [hunk 9]
```diff
@@ -81,6 +82,38 @@ const UsersManagement: React.FC = () => {
             case UsersManagementTableHeadersNames.DESK: {
                 return row[cellName] ? row[cellName] : noDeskAssignment
             }
+            case UsersManagementTableHeadersNames.SOURCE_ORGANIZATION : {
+                return (
+                    <Select
+                        MenuProps={{
+                            anchorOrigin: {
```
Comment (MayaSasson) [inline client/src/components/App/Content/UsersManagement/UsersManagement.tsx:89]: Single quotes

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx [hunk 10]
```diff
@@ -81,6 +82,38 @@ const UsersManagement: React.FC = () => {
...
+                            transformOrigin: {
+                                vertical: "top",
+                                horizontal: "left"
+                            },
+                            getContentAnchorEl: null
+                        }}
+                        variant='outlined'
+                        className={classes.sourceOrganization}
+                        label={sourceOrganizationLabel}
+                        value={row[cellName]}
```
Comment (NaamanTsur) [inline client/src/components/App/Content/UsersManagement/UsersManagement.tsx:102]: You know the value is the source organization. I think it's better to be explicit: `row.sourceOrganization` or what it's name is.

**Review by NaamanTsur — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx (hunk 9 above)
Comment (nevobonneh) [reply client/src/components/App/Content/UsersManagement/UsersManagement.tsx:89]: fixed

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx (hunk 10 above)
Comment (nevobonneh) [reply client/src/components/App/Content/UsersManagement/UsersManagement.tsx]: fixed

**Review by NaamanTsur — APPROVED** [review]

**Review by ItayBenMoshe — APPROVED** [review]

**Pull request merged**